### PR TITLE
Source Slack: add unexpected fields

### LINKED ISSUE #24499: Source Slack: Add undeclared columns to spec
## What
In this [PR](https://github.com/airbytehq/airbyte/pull/23985), `BasicReadTest.test_read` test in the Connector Acceptance Tests was updated to fail if the connector produces
stream records which contain columns that haven't been declared in the spec.

Slack currently fails the updated test. Its `acceptance-test-config.yml` was edited with the
`fail_on_extra_columns: false` parameter in order to avoid this change from making the connector fail CAT.

We want to add the undeclared columns to the spec.

**How this will help**:
- Column selection is currently blocked for connectors that don't declare all columns that they pass. This
  is because turning on column selection will stop these undeclared columns from being sent, when they were
  previously being sent. Doing this allows us to enable column selection for Slack!
- Users will better understand the shape of the data they'll receive, since the columns will match the spec.

## How
The following descriptions of streams that pass undeclared columns come from results of the failed connector acceptance test:

```
columns.txt:./source-slack.txt-81-The channels stream has the following schema errors:
columns.txt:./source-slack.txt:82:Additional properties are not allowed ('parent_conversation', 'updated', 'pending_connected_team_ids', 'shared_team_ids', 'context_team_id' were unexpected)
columns.txt:./source-slack.txt-158-The threads stream has the following schema errors:
columns.txt:./source-slack.txt:159:Additional properties are not allowed ('subtype' was unexpected)
columns.txt:./source-slack.txt-194-The users stream has the following schema errors:
columns.txt:./source-slack.txt:195:Additional properties are not allowed ('who_can_share_contact_card', 'is_email_confirmed' were unexpected)

```

1. Add the missing properties indicated by the `Additional properties are not allowed ('<column>', 'column' were unexpected)` logs to the connector's spec.
2. Remove `fail_on_extra_columns: false` from the `acceptance-test-config.yml` file.
3. Commit changes to spec and `acceptance-test-config.yml` and open a PR.
4. Run tests on the connector, either automatically via CI or manually via the `/test` command
5. Profit!

Definition of done: Slack passes CAT without declaring `fail_on_extra_columns: false`. If the
API starts sending over extra columns in the future, we will catch and fix them as part of the #connector-health
movement.

### PR DESCRIPTION
**Pull request by arsenlosenko**
## What
Add unexpected fields in scope of this issue:
https://github.com/airbytehq/airbyte/issues/24499

## How
* add field `is_locked` to stream `channel_messages`
* add fields `context_team_id`, `updated`, `shared_team_ids`, `parent_conversation`, `pending_connected_team_ids` to stream `channels`
* add fields `subtype`, `is_locked`, `last_read`, `parent_user_id` to stream `threads`
* add fields `is_forgotten`, `is_invited_user`, `is_email_confirmed`, `who_can_share_contact_card` to stream `users`

### CODE REVIEW

File: docs/integrations/sources/slack.md [hunk 1]
```diff
@@ -136,6 +136,7 @@ It is recommended to sync required channels only, this can be done by specifying
 
 | Version | Date       | Pull Request                                             | Subject                                                     |
 |:--------|:-----------|:---------------------------------------------------------|:------------------------------------------------------------|
+| 0.3.0  | 2023-09-18 | [30521](https://github.com/airbytehq/airbyte/pull/30521) | Add unexpected fields to streams `channel_messages`, `channels`, `threads`, `users`         |
```
Comment (davydov-d) [inline docs/integrations/sources/slack.md:139]: please format the table

File: airbyte-integrations/connectors/source-slack/source_slack/schemas/channels.json [hunk 2]
```diff
@@ -17,6 +17,9 @@
     "is_im": {
       "type": "boolean"
     },
+    "context_team_id": {
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-slack/source_slack/schemas/channels.json:20]: should we update the expected records for the streams where the schema is updated?

File: airbyte-integrations/connectors/source-slack/source_slack/schemas/channels.json [hunk 3]
```diff
@@ -105,6 +114,12 @@
     "num_members": {
       "type": "integer"
     },
+    "parent_conversation": {
+      "type": ["null", "string"]
+    },
+    "pending_connected_team_ids": {
+      "type": "array"
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-slack/source_slack/schemas/channels.json:121]: can you please describe the array items structure?

**Review by davydov-d — CHANGES_REQUESTED** [review]
I think we should also remove `fail_on_extra_columns: false` in the CAT config, shouldn't we?

File: docs/integrations/sources/slack.md (hunk 1 above)
Comment (arsenlosenko) [reply docs/integrations/sources/slack.md]: formatted this table

File: airbyte-integrations/connectors/source-slack/source_slack/schemas/channels.json (hunk 3 above)
Comment (arsenlosenko) [reply airbyte-integrations/connectors/source-slack/source_slack/schemas/channels.json]: specified type of items in array fields 👍

**Review by davydov-d — APPROVED** [review]

File: airbyte-integrations/connectors/source-slack/source_slack/schemas/channels.json (hunk 2 above)
Comment (arsenlosenko) [reply airbyte-integrations/connectors/source-slack/source_slack/schemas/channels.json:20]: updated expectred records

**Review by lazebnyi — APPROVED** [review]

**Pull request merged**